File: backend/Multibanca.Domain.Models/Multibanca/BBVA/CargarDocumentosCliente.cs
using Common.Domain.Models;

namespace Multibanca.Domain.Models.Multibanca.BBVA;

/// <summary>
/// Registro persistente de la confirmacion documental realizada en la actividad Cargar Documentos Cliente.
/// </summary>
public class cargar_documentos_cliente : base_auditoria
{
    public long id { get; set; }
    public long id_expediente { get; set; }
    public string id_actividad { get; set; } = "BBVA_CONTACTO_CARGAR_DOCUMENTOS_CLIENTE_CBF7A738";
    public bool documentos_adjuntos { get; set; }
    public string? observaciones { get; set; }
}

File: frontend/src/features/actividades/cargar_documentos_cliente/models/cargar_documentos_cliente.ts
/**
 * Contrato frontend/backend del registro de confirmacion documental de Cargar Documentos Cliente.
 */
export interface CargarDocumentosCliente {
  id?: number;
  id_expediente: number;
  id_actividad: string;
  documentos_adjuntos: boolean;
  observaciones?: string | null;
  is_active?: boolean;
  row_status?: boolean;
}

/**
 * Informacion general que la pantalla externa de Cargar Documentos Cliente puede mostrar enmascarada.
 */
export interface CargarDocumentosClienteInfo {
  /** Identificador del expediente presentado como folio Presto. */
  id_expediente: number;
  /** Nombre del cliente obtenido desde la informacion validada. */
  nombre_completo_t1?: string | null;
  nombre_cliente?: string | null;
  correo_declarativo?: string | null;
  telefono_declarativo?: string | null;
  /** Usuario responsable del workflow presentado como analista de vivienda. */
  nombre_analista?: string | null;
}

/**
 * Crea el estado inicial del formulario de confirmacion documental para un expediente.
 *
 * @param id_expediente Identificador del expediente en Presto.
 * @returns Modelo inicial de Cargar Documentos Cliente con la actividad fija `ACT_DOCS_CLIENTE`.
 */
export const EMPTY_CARGAR_DOCUMENTOS_CLIENTE = (
  id_expediente: number,
): CargarDocumentosCliente => ({
  id_expediente,
  id_actividad: 'BBVA_CONTACTO_CARGAR_DOCUMENTOS_CLIENTE_CBF7A738',
  documentos_adjuntos: false,
  observaciones: '',
  is_active: true,
  row_status: true,
});
